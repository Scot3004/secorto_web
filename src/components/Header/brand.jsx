/** @jsx jsx */
import { jsx } from 'theme-ui'

import HomePageLink from './homepage-link'

export default ({title}) => (
  <div
    sx={{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      padding: 0,
    }}
  >
    <HomePageLink />
    <span sx={{ ml: 2, fontSize: [42], color: 'primary' }}>{title}</span>
  </div>
)
