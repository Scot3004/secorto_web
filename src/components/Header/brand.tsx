/** @jsxImportSource theme-ui */
import HomePageLink from "./homepage-link"

const Brand = ({ title }) => (
  <div
    sx={{
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      padding: 0,
    }}
  >
    <HomePageLink />
    <span sx={{ ml: 2, fontSize: [42], color: "primary" }}>{title}</span>
  </div>
)

export default Brand
