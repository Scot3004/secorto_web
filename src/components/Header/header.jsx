/** @jsx jsx */
import { jsx } from 'theme-ui'

import HomePageLink from './homepage-link'

export default ({ children }) => {
  return (
    <header
      sx={{
        display: 'flex',
        alignItems: 'center',
        boxShadow: theme => `0 0 8px  ${theme.colors.primary}`,
        minHeight: [58]
      }}
    >
      <div sx={{ mx: 'auto' }} />

      <div
        sx={{
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
          padding: 0,
        }}
      >
        <HomePageLink />
        <span sx={{
          ml: 2,
          fontSize: [42],
          color: 'primary',
          fontFamily: 'title',
          position: 'relative',
          top: '5px'
        }}>{children}</span>
      </div>
      <div sx={{ mx: 'auto' }} />
    </header>
  )
}
