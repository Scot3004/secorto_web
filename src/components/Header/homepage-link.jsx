/** @jsx jsx */
import { Link } from 'gatsby'
import { jsx } from 'theme-ui'

import Logo from '../../assets/logo.svg'

export default () => (
  <div>
    <Link
      to="/"
    >
      <img
        src={Logo}
        alt="logo"
        sx={{
          width: [36],
          height: [36],
          verticalAlign: 'middle',
          borderRadius: "50%",
          transition: "box-shadow .5s",
          boxShadow: theme => `2px 2px 8px ${theme.colors.primary}`,
          ":hover": {
            boxShadow: theme => `2px 2px 8px ${theme.colors.hover}`,
          },
        }}></img>
    </Link>
  </div>
)
