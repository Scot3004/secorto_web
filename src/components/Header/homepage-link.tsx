/** @jsxImportSource theme-ui */
import { Link } from "gatsby"
import Logo from "../../assets/logo.inline.svg"

const HomepageLink = () => (
  <div>
    <Link to="/">
      <Logo
        alt="logo"
        sx={{
          width: [36],
          height: [36],
          verticalAlign: "middle",
          borderRadius: "50%",
          transition: "box-shadow .5s",
          boxShadow: theme => `2px 2px 8px ${theme.colors.primary}`,
          ":hover": {
            boxShadow: theme => `2px 2px 8px ${theme.colors.hover}`,
          },
        }}
      />
    </Link>
  </div>
)

export default HomepageLink
