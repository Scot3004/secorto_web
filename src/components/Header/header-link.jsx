/** @jsx jsx */
import { Link } from 'gatsby'
import { jsx } from 'theme-ui'
import Header from './header'

export default ({children, ...props}) => (
  <Header>
    <Link {...props}
      sx={{
        color: 'primary',
        textDecoration: "none",
        ":hover": {
          color: "hover"
        }
      }}
    >
      {children}
    </Link>
  </Header>
)
