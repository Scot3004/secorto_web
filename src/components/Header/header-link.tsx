/** @jsxImportSource theme-ui */
import { Link } from "gatsby"
import Header from "./header"

const HeaderLink = ({ children, to, ...props }) => (
  <Header>
    <Link
      to={to}
      {...props}
      sx={{
        color: "primary",
        transition: "all 0.3s linear",
        textDecoration: "none",
        ":hover": {
          color: "hover",
        },
      }}
    >
      {children}
    </Link>
  </Header>
)

export default HeaderLink
