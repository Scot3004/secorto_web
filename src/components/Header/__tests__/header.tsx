import { ThemeProvider } from "@theme-ui/theme-provider"
import * as React from "react"
import renderer from "react-test-renderer"
import Header from "../header"

const theme = {
  colors: {
    primary: "red",
  },
}

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Header>header content</Header>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
