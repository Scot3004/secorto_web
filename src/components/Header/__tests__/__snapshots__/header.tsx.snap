// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 0 8px var(--theme-ui-colors-primary);
  min-height: 58px;
}

.emotion-1 {
  margin-left: auto;
  margin-right: auto;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 0;
}

.emotion-3 {
  width: 36px;
  height: 36px;
  vertical-align: middle;
  border-radius: 50%;
  -webkit-transition: box-shadow .5s;
  transition: box-shadow .5s;
  box-shadow: 2px 2px 8px var(--theme-ui-colors-primary);
}

.emotion-3:hover {
  box-shadow: 2px 2px 8px undefined;
}

.emotion-4 {
  margin-left: 8px;
  font-size: 42px;
  color: var(--theme-ui-colors-primary);
  font-family: title;
  position: relative;
  top: 5px;
  margin: 0 1rem;
  font-weight: 400;
}

<header
  className="emotion-0"
>
  <div
    className="emotion-1"
  />
  <div
    className="emotion-2"
  >
    <div>
      <a
        href="/"
      >
        <test-file-stub
          alt="logo"
          className="emotion-3"
        />
      </a>
    </div>
    <span
      className="emotion-4"
    >
      header content
    </span>
  </div>
  <div
    className="emotion-1"
  />
</header>
`;
